src/include/utilities.h

```diff
@@ -106,6 +106,30 @@ FILE *fstream_tee(FILE *orig, FILE *copy);
     pthread_cleanup_pop(0); \
 }
 
+typedef sem_t *pshared_sem_t;
+static inline int pshared_sem_post(pshared_sem_t sem)
+{
+    return sem_post(sem);
+}
+static inline int pshared_sem_wait(pshared_sem_t sem)
+{
+    return sem_wait(sem);
+}
+
+/* macOS doesn't support sem_init(), so use Mach semaphores instead */
+#ifdef __APPLE__
+#include <mach/mach_init.h>
+#include <mach/task.h>
+#include <mach/semaphore.h>
+#undef PAGE_SIZE
```

Comment:
Maybe ifndef?

Reply:
not sure how? The problem here is that mach headers redefine PAGE_SIZE to
```#define PAGE_SIZE               vm_page_size```
and dosemu assumes PAGE_SIZE is constant.

Reply:
What is vm_page_size?
Still looks constant to me, or it would
be `vm_page_size()` instead?

Reply:
it's an extern variable:
`extern  vm_size_t       vm_page_size;`

Reply:
```
mapping.c:92:23: error: variable length array declaration not allowed at file scope
static unsigned char *lowmem_aliasmap[ALIAS_SIZE/PAGE_SIZE];
                      ^               ~~~~~~~~~~~~~~~~~~~~
```

Reply:
We have this in cpu.h:
```
#ifndef PAGE_SIZE
#define PAGE_SIZE       4096
#endif
```
Maybe just prepend that with
```
#ifdef __APPLE__
#undef PAGE_SIZE
#endif
```
?

Reply:
that doesn't make the error go away in mapping.c: cpu.h is often included via emu.h before utilities.h. utilities.h then overwrites PAGE_SIZE via the mach headers.

Reply:
So mach header also do stray `#undef`?
How nasty.